src/dune_rules/module_compilation.ml

```diff
@@ -199,6 +194,21 @@ let build_cm cctx ~precompiled_cmi ~cm_kind (m : Module.t)
     |> List.concat_map ~f:(fun p ->
            [ Command.Args.A "-I"; Path (Path.build p) ])
   in
+  let melange_args =
+    match cm_kind with
+    | Melange Cmj ->
+      let pkg_name_args =
+        match CC.package cctx with
+        | None -> []
+        | Some pkg ->
+          [ "--bs-package-name"; Package.Name.to_string (Package.name pkg) ]
+      in
+      "--bs-stop-after-cmj" :: "--bs-package-output"
+      :: (* This should prob be Path.t or Path.Source.t *)
+         Path.Build.to_string (CC.dir cctx)
```

Comment:
Instead of stringifying the path, you should pass it with the `Command.Args.Path` constructor

Reply:
Fixed in https://github.com/ocaml/dune/pull/6349/commits/b0c211dfc2cbea2f3c9acb91e385108a85ce4c5e.

As a side effect, due to usage of `Path.build` to build `Path.t` values, we got rid of the `_build/default` prefixes for package names that [I was asking about](https://github.com/ocaml/dune/pull/6349#discussion_r1008689179) a few days ago 🎉  (see the tests changes on the linked commit)